leetcode/string_2.rs

```diff
@@ -0,0 +1,36 @@
+//Sorting the sentence
+pub fn sort_sentence(s: String) -> String {
+    let mut firstvector: Vec<&str> = s.split(" ").collect::<Vec<&str>>();
+
+    firstvector.sort_by(|a, b| a.chars().last().unwrap().cmp(&(b.chars().last().unwrap())));
+
+    let newindex = firstvector
+        .iter()
+        .map(|x| String::from(x.split_at(x.len() - 1).0));
+
+    let finalvector: Vec<String> = newindex.collect();
+
+    finalvector.join(" ")
+}
+
+//Count Items Matching a Rule
+pub fn count_matches(items: Vec<Vec<String>>, rule_key: String, rule_value: String) -> i32 {
+    let mut matches = 0;
+
+    let x: &str = &rule_key;
+
+    let i = match x {
+        "type" => 0,
+        "color" => 1,
+        "name" => 2,
+        _ => return 0,
+    };
+
+    for j in items {
```

Comment:
Try to replace this `for`-loop with an iterator (`items.iter()`). Using [`filter`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter) would make the intention of the code clearer.